feature/chatting/src/commonMain/kotlin/com/nexters/emotia/feature/chatting/ChattingViewModel.kt

```diff
@@ -1,7 +1,129 @@
+
+    var uiState by mutableStateOf(ChattingUiState())
+        private set
+
+    init {
+        createChatRoom()
+    }
+
+    private fun createChatRoom() {
+        viewModelScope.launch {
+            uiState = uiState.copy(isLoading = true, error = null)
+
+            chattingRepository.createRoom("유저네임하드코딩해야됨각자")
+                .onSuccess { chatRoom ->
+                    val firstMessage = ChatMessage(
+                        text = chatRoom.firstMessage,
+                        type = BubbleType.OTHER
+                    )
+
+                    uiState = uiState.copy(
+                        roomId = chatRoom.roomId,
+                        messages = listOf(firstMessage),
+                        isLoading = false,
+                        showEmotionChips = true,
+                        emotionOptions = getDefaultEmotions()
+                    )
+                }
+                .onFailure { exception ->
+                    uiState = uiState.copy(
+                        isLoading = false,
+                        error = "채팅룸 생성에 실패했습니다: ${exception.message}"
+                    )
+                }
+        }
+    }
+
+    fun onInputTextChanged(text: String) {
+        uiState = uiState.copy(currentInputText = text)
+    }
+
+    fun onSendMessage() {
+        if (uiState.currentInputText.isBlank() || uiState.roomId == null) return
+
+        val userMessage = ChatMessage(
+            text = uiState.currentInputText,
+            type = BubbleType.MINE
+        )
+
+        val messageToSend = uiState.currentInputText
+        val roomId = uiState.roomId!!
+
+        uiState = uiState.copy(
+            messages = uiState.messages + userMessage,
+            currentInputText = "",
+            isLoading = true
+        )
+
+        sendChatMessage(roomId, messageToSend)
+    }
+
+
+    private fun sendChatMessage(roomId: Int, message: String) {
+        viewModelScope.launch {
+            // chatting typing indicator 를 보기 위해 딜레이 걸어둠
+            kotlinx.coroutines.delay(1000)
+
+            chattingRepository.sendChat(roomId, message)
+                .onSuccess { chatResponse ->
+                    val aiResponse = ChatMessage(
+                        text = chatResponse.message,
+                        type = BubbleType.OTHER
+                    )
+
+                    uiState = uiState.copy(
+                        messages = uiState.messages + aiResponse,
+                        isLoading = false,
+                        error = null
+                    )
+                }
+                .onFailure { exception ->
+                    uiState = uiState.copy(
+                        isLoading = false,
+                        error = "메시지 전송에 실패했습니다: ${exception.message}"
+                    )
+                }
+        }
+    }
+
+    private fun getDefaultEmotions() = listOf(
+        EmotionOption(text = "기쁨"),
+        EmotionOption(text = "슬픔"),
+        EmotionOption(text = "화남"),
+    )
```

Comment:
```suggestion
    private val defaultEmotions() = listOf(
        EmotionOption(text = "기쁨"),
        EmotionOption(text = "슬픔"),
        EmotionOption(text = "화남"),
    )
```

메모리 관점에서 불변상수로 두는 것이 성능적으로 좋을 것 같습니다